src/script/components/LexicalInput/components/SendMessageButton/SendMessageButton.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {useEffect} from 'react';
+
+import {useLexicalComposerContext} from '@lexical/react/LexicalComposerContext';
+import cx from 'classnames';
+import {$getRoot, $nodesOfType, COMMAND_PRIORITY_LOW, KEY_ENTER_COMMAND, type LexicalEditor} from 'lexical';
+
+import {Icon} from 'Components/Icon';
+import {t} from 'Util/LocalizerUtil';
+
+import {User} from '../../../../entity/User';
+import {MentionEntity} from '../../../../message/MentionEntity';
+import {BeautifulMentionNode} from '../../nodes/MentionNode';
+
+const createMentionEntity = (user: Pick<User, 'id' | 'name' | 'domain'>, mentionPosition: number): MentionEntity => {
+  const userName = user.name();
+  const mentionLength = userName.length + 1;
+
+  return new MentionEntity(mentionPosition, mentionLength, user.id, user.domain);
+};
+
+interface SendMessageButtonProps {
+  textValue: string;
+  onSend: any;
```

Comment:
why are we not passing specific type?

`type OnSendFunction = (text: string) => void;`

do you know why onSend function returns 2 different types?

 const onSend = (text: string): void | boolean